tests/test_config.py

```diff
@@ -2388,6 +2400,172 @@ def test_model_type_llm(self):
             config.inference_output_fields, DEFAULT_LLM_INFERENCE_OUTPUT_FIELDS
         )
 
```

Comment:
Also, dang, there are a lot of tests in this file! This is likely a reach here, but I wanted to drop the idea of [parameterization](https://docs.pytest.org/en/7.1.x/example/parametrize.html) in case it's possible here. It may not be, but wanted to share in case we can use it in more testing in the future.


This comment is a conversational comment, it's not a review/blocking comment.

Note: I dropped a Pytest link (and switching to it should be as simple as changing the command line command from `unittest` to `pytest`, but [unittest also seems to have options](https://stackoverflow.com/questions/32899/how-do-you-generate-dynamic-parameterized-unit-tests-in-python) if we prefer to not go the PyTest route.